src/post_strategy.ts

```diff
@@ -0,0 +1,44 @@
+import { APIStrategy } from "./api_strategy";
+import axios from 'axios';
+import { URLSearchParams } from "url";
+var _ = require('lodash');
+
+/**
+ * The API Strategy for post calls
+ */
+export class PostStrategy extends APIStrategy {
+    protected async getRawResponse(token: string) {
+        // Check api shared cache first
+        let cache = APIStrategy.getCache();
+        let cacheResult = cache.getCachedValue(this.getCacheKey(token));
+        if (cacheResult) {
+            console.log(this.apiJSON.name + " accessed cache at key " + this.getCacheKey(token));
```

Comment:
May want to move cache error handling to the caching class/implementation.

Reply:
Sorry this was just left over logging on my part, I know that people usually use a logger class, but I have never looked into the best practices for logging, so I would also be interested in knowing if there is a specific practice we should use.